src/tiled/propertiesdock.cpp

```diff
@@ -293,6 +296,42 @@ void PropertiesDock::renameProperty(const QString &name)
     undoStack->push(new RenameProperty(mMapDocument, mMapDocument->currentObjects(), oldName, name));
 }
 
+void PropertiesDock::showContextMenu(const QPoint& pos)
+{   
+    QtBrowserItem *item = mPropertyBrowser->currentItem();
+    if (!mPropertyBrowser->isCustomPropertyItem(item))
+        return;
+    QPoint globalPos = mPropertyBrowser->mapToGlobal(pos);
+
+    QString name = item->property()->propertyName();
+    Object *object = mMapDocument->currentObject();
+    QVariant value = object->property(name);
+
+    QMenu *contextMenu;
+    contextMenu = new QMenu(mPropertyBrowser);
+    contextMenu->addSection(tr("Convert to:"));
+
+    QList<unsigned int> convertTo;
+    convertTo << QVariant::Bool << QVariant::Color << QVariant::Double << filePathTypeId() << QVariant::Int << QVariant::String;
+    foreach (const unsigned int &toType, convertTo){
+        if (value.canConvert(toType) && (value.type() != toType)) {
+            QAction* action = contextMenu->addAction(typeToName(toType));
+            action->setData(toType);
+        }
+    }
+
+    QAction* selectedItem = contextMenu->exec(globalPos);
+    if (selectedItem) {
+        unsigned int selectedType = selectedItem->data().toInt();
+        if (value.canConvert(selectedType)) {
+            value.convert(selectedType);
+            removeProperty();
+            addProperty(name, value);
+        }
+    }
+    delete contextMenu;
```

Comment:
Please allocate the `QMenu contextMenu` on the stack instead of on the heap, then you can drop this line.